File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/ssm/ParameterName.java
package uk.nhs.nhsx.core.aws.ssm;

import uk.nhs.nhsx.core.ValueType;

public class ParameterName extends ValueType<ParameterName> {

    protected ParameterName(String value) {
        super(value);
    }

    public static ParameterName of(String value) {
        return new ParameterName(value);
    }

    public ParameterName withPrefix(String prefix) {
        return ParameterName.of(prefix + "/" + value);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/ssm/ParameterName.kt
package uk.nhs.nhsx.core.aws.ssm

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class ParameterName private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<ParameterName>(::ParameterName)

    fun withPrefix(prefix: String) = of("$prefix/$value")
}
